api/src/main/java/org/apache/iceberg/Table.java

```diff
@@ -44,6 +44,28 @@
    */
   TableScan newScan();
 
+
+  /**
+   * Create a table scan which can read append data from {@param fromSnapshotId}
+   * exclusive and up to {@param toSnapshotId} inclusive
+   * @param fromSnapshotId - the last snapshot id read by the user, exclusive
+   * @param toSnapshotId - read append data up to this snapshot id
+   * @return a table scan which can read append data from {@param fromSnapshotId}
+   * exclusive and up to {@param toSnapshotId} inclusive
+   */
+  TableScan appendsBetween(long fromSnapshotId, long toSnapshotId);
+
+  /**
+   * Create a table scan which can read append data from {@param fromSnapshotId}
+   * exclusive and up to the current snapshot id exclusive
+   * @param fromSnapshotId - the last snapshot id read by the user, exclusive
+   * @return a table scan which can read append data from {@param fromSnapshotId}
+   * exclusive and up to current snapshot inclusive
+   */
+  default  TableScan appendsAfter(long fromSnapshotId) {
+    return appendsBetween(fromSnapshotId, currentSnapshot().snapshotId());
```

Comment:
This should check whether `currentSnapshot` is null, or else it will cause a `NullPointerException` for new tables.